### PR TITLE
fix(diagnostic): get border from config

### PR DESCRIPTION
**Pull request by glepnir**
Problem: border value is hardcoded

Solution: get it from config interface

### CODE REVIEW

File: runtime/lua/vim/diagnostic.lua [hunk 1]
```diff
@@ -152,6 +152,8 @@ local M = {}
 --- @field suffix? string|table|(fun(diagnostic:vim.Diagnostic,i:integer,total:integer): string, string)
 ---
 --- @field focus_id? string
+---
+--- @field border? string
```
Comment (justinmk) [inline runtime/lua/vim/diagnostic.lua:156]: is there a generic float class that `vim.diagnostic.Opts.Float` can inherit from?

File: runtime/lua/vim/_defaults.lua [hunk 2]
```diff
@@ -190,7 +190,10 @@ do
     })
 
     vim.keymap.set('n', '<C-W>d', function()
-      vim.diagnostic.open_float({ border = 'rounded' })
+      local config = vim.diagnostic.config()
+      local border = type(config.float) == 'table' and (config.float.border or 'rounded')
+        or 'rounded'
```
Comment (justinmk) [inline runtime/lua/vim/_defaults.lua:195]: could use `vim.tbl_get`.  

side-note:  should `vim.tbl_get` allow non-table input? that would make it more flexible, but less strict.

```suggestion
      local border = vim.tbl_get(config, 'float', 'border') or 'rounded'
```

Comment (glepnir) [reply runtime/lua/vim/_defaults.lua:195]: forget some util functions someitme .......

File: runtime/lua/vim/diagnostic.lua (hunk 1 above)
Comment (glepnir) [reply runtime/lua/vim/diagnostic.lua:156]: just search float in help doc seems like we don't have a float class  ?

Comment (justinmk) [reply runtime/lua/vim/diagnostic.lua:156]: yeah i don't see one in `runtime/lua/vim/_meta/api.lua`

**Comment by mfussenegger** [comment]
Would it be an option to have a global `float_border` option? The border is also a frequent config option for plugins and in some LSP methods. It would be great if a user would only have to configure it once instead of per module/functionality to have consistent borders

**Comment by justinmk** [comment]
> Would it be an option to have a global `float_border` option?

a 'winoptions' or 'winstyle' option is probably needed at some point. 

but this PR seems good meanwhile, because this mapping is for diagnostics specifically.

**Comment by gpanders** [comment]
> Would it be an option to have a global `float_border` option? The border is also a frequent config option for plugins and in some LSP methods. It would be great if a user would only have to configure it once instead of per module/functionality to have consistent borders

Related: `vim.ui.config()` https://github.com/neovim/neovim/issues/25514

File: runtime/lua/vim/_defaults.lua [hunk 3]
```diff
@@ -190,7 +190,9 @@ do
     })
 
     vim.keymap.set('n', '<C-W>d', function()
-      vim.diagnostic.open_float({ border = 'rounded' })
+      local config = vim.diagnostic.config()
+      local border = vim.tbl_get(config, 'float', 'border') or 'rounded'
+      vim.diagnostic.open_float({ border = border })
```
Comment (gpanders) [inline runtime/lua/vim/_defaults.lua:195]:
```suggestion
      vim.diagnostic.open_float()
```

**Review by justinmk — APPROVED** [review]

File: runtime/doc/diagnostic.txt [hunk 4]
```diff
@@ -499,6 +499,7 @@ Lua module: vim.diagnostic                                    *diagnostic-api*
                           diagnostic instead of prepending it. Overrides the
                           setting from |vim.diagnostic.config()|.
       • {focus_id}?       (`string`)
+      • {border}?         (`string`)
```
Comment (gpanders) [inline runtime/doc/diagnostic.txt:502]: Can we link to nvim_open_win so that users know what valid values for this option are?

Comment (glepnir) [reply runtime/doc/diagnostic.txt:502]: before i tried add a see section. but i don't saw it in hover idk.

File: runtime/doc/diagnostic.txt [hunk 5]
```diff
@@ -499,6 +499,7 @@ Lua module: vim.diagnostic                                    *diagnostic-api*
                           diagnostic instead of prepending it. Overrides the
                           setting from |vim.diagnostic.config()|.
       • {focus_id}?       (`string`)
+      • {border}?         (`string`) see |vim.api.nvim_open_win|.
```
Comment (justinmk) [inline runtime/doc/diagnostic.txt:502]: not a valid tag. that's why docs ci is failing
```suggestion
      • {border}?         (`string`) see |vim.api.nvim_open_win()|.
```

**Review by gpanders — APPROVED** [review]

**Pull request merged**